spec/datadog/open_feature/remote_spec.rb

```diff
@@ -0,0 +1,201 @@
+# frozen_string_litral: true
+
+require 'spec_helper'
+require 'datadog/open_feature/remote'
+require 'datadog/core/remote/configuration/repository'
+
+RSpec.describe Datadog::OpenFeature::Remote do
+  let(:remote) { described_class }
+  let(:telemetry) { instance_double(Datadog::Core::Telemetry::Component) }
+  let(:receivers) { remote.receivers(telemetry) }
+  let(:receiver) { receivers[0] }
+
+  describe '.capabilities' do
+    it { expect(remote.capabilities).to eq([70368744177664]) }
+  end
+
+  describe '.products' do
+    it { expect(remote.products).to eq(['FFE_FLAGS']) }
+  end
+
+  describe '.receivers' do
+    it 'returns receivers' do
+      expect(receivers).to have(1).element
+      expect(receiver).to be_a(Datadog::Core::Remote::Dispatcher::Receiver)
+    end
+
+    it 'matches FFE_FLAGS product paths' do
+      path = Datadog::Core::Remote::Configuration::Path.parse('datadog/1/FFE_FLAGS/ufc-test/config')
+
+      expect(receiver.match?(path)).to be(true)
+    end
+  end
+
+  describe 'receiver logic' do
+    before do
+      allow(telemetry).to receive(:error)
+      allow(Datadog::OpenFeature).to receive(:engine).and_return(engine)
+    end
+
+    let(:engine) { Datadog::OpenFeature::EvaluationEngine.new(reporter, telemetry: telemetry) }
+    let(:reporter) { instance_double(Datadog::OpenFeature::Exposures::Reporter) }
+    let(:repository) { Datadog::Core::Remote::Configuration::Repository.new }
+    let(:target) do
+      Datadog::Core::Remote::Configuration::Target.parse(
+        {
+          'custom' => {'v' => 1},
+          'hashes' => {'sha256' => Digest::SHA256.hexdigest(content_data)},
+          'length' => content_data.length
+        }
+      )
+    end
+    let(:content) do
+      Datadog::Core::Remote::Configuration::Content.parse(
+        {
+          path: 'datadog/1/FFE_FLAGS/latest/config',
+          content: StringIO.new(content_data)
+        }
+      )
+    end
+    let(:content_data) do
+      <<~JSON
+        {
+          "data": {
+            "type": "universal-flag-configuration",
+            "id": "1",
+            "attributes": {
+              "createdAt": "2024-04-17T19:40:53.716Z",
+              "format": "SERVER",
+              "environment": { "name": "test" },
+              "flags": {
+                "test_flag": {
+                  "key": "test_flag",
+                  "enabled": true,
+                  "variationType": "STRING",
+                  "variations": {
+                    "control": { "key": "control", "value": "control_value" }
+                  },
+                  "allocations": [
+                    {
+                      "key": "rollout",
+                      "splits": [{ "variationKey": "control", "shards": [] }],
+                      "doLog": false
+                    }
+                  ]
+                }
+              }
+            }
+          }
+        }
```

Comment:
I recently noticed that [the config format used in system-tests](https://github.com/DataDog/system-tests/blob/main/tests/parametric/test_feature_flag_exposure/flags-v1.json) is different from the one we've been using

I assume that this is the format that we'll need to support for the system tests, so I've been updating https://github.com/DataDog/dd-trace-rb/pull/5022 accordingly

```suggestion
        {
          "createdAt": "2024-04-17T19:40:53.716Z",
          "format": "SERVER",
          "environment": { "name": "test" },
          "flags": {
            "test_flag": {
              "key": "test_flag",
              "enabled": true,
              "variationType": "STRING",
              "variations": {
                "control": { "key": "control", "value": "control_value" }
              },
              "allocations": [
                {
                  "key": "rollout",
                  "splits": [{ "variationKey": "control", "shards": [] }],
                  "doLog": false
                }
              ]
            }
          }
        }
```